pkg/cluster/cluster.go

```diff
@@ -666,9 +686,25 @@ func getKubeadmResetOptions(kubeadmVersionStr string) (string, error) {
 	return kubeadmResetOptions, nil
 }
 
-func applyNetworkPlugin(machineName string, outWriter io.Writer) error {
-	_, err := machinectl.RunCommand(outWriter, nil, "", "shell", machineName, "/usr/bin/kubectl", "apply", "-f", weaveNet)
-	return err
+func applyNetworkPlugin(machineName string, cniModel string, outWriter io.Writer) error {
+	switch cniModel {
+	case "weave":
+		_, err := machinectl.RunCommand(outWriter, nil, "", "shell", machineName, "/usr/bin/kubectl", "apply", "-f", weaveNet)
+		return err
+	case "flannel":
+		_, err := machinectl.RunCommand(outWriter, nil, "", "shell", machineName, "/usr/bin/kubectl", "apply", "-f", flannelNet)
+		return err
+	case "calico":
+		_, err1 := machinectl.RunCommand(outWriter, nil, "", "shell", machineName, "/usr/bin/kubectl", "apply", "-f", calicoNet1)
+		_, err2 := machinectl.RunCommand(outWriter, nil, "", "shell", machineName, "/usr/bin/kubectl", "apply", "-f", calicoNet2)
+		if err1 != nil {
+			return err1
+		}
+		return err2
+	default:
+		log.Fatal("Incorrect cni model")
```

Comment:
All code in `pkg` should be usable as a library, i.e. please do not `log.Fatal` here but return an error that can be handled by the caller.